File: src/II045.py
# Definition for a binary tree node.
from queue import Queue


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def findBottomLeftValue(self, root: TreeNode) -> int:
        qq = Queue()
        qq.put((root, 1))
        dep = 0
        ret = 0
        while not qq.empty():
            (cur, cur_dep) = qq.get()
            if cur_dep > dep:
                dep = cur_dep
                ret = cur.val
            if cur.left:
                qq.put((cur.left, cur_dep+1))
            if cur.right:
                qq.put((cur.right, cur_dep+1))
        return ret

File: src/II045.java
import java.util.LinkedList;
import java.util.Queue;

public class II045 {

}

// Definition for a binary tree node.
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public int findBottomLeftValue(TreeNode root) {
        Queue<TreeNode> qq = new LinkedList<>();
        qq.offer(root);
        int ret = 0;
        while (!qq.isEmpty()) {
            var cur = qq.poll();
            ret = cur.val;
            if (cur.right != null) {
                qq.offer(cur.right);
            }
            if (cur.left != null) {
                qq.offer(cur.left);
            }
        }
        return ret;
    }
}
